weld/llvm.rs

```diff
@@ -1097,6 +1097,56 @@ impl LlvmGenerator {
     *********************************************************************************************/
 
 
+    fn gen_minmax(&mut self, ll_ty: &String,
+                  op: &BinOpKind,
+                  left_tmp: &String,
+                  right_tmp: &String,
+                  output_tmp: &String,
+                  ty: &Type,
+                  func: &SirFunction,
+                  ctx: &mut FunctionContext) -> WeldResult<()> {
+        use super::ast::BinOpKind::*;
+        
+        match *ty {
+            Scalar(s) | Simd(s) => {
+                if s.is_signed_integer() | s.is_unsigned_integer() {
+                    let sel_tmp = ctx.var_ids.next();
+
+                    match op {
```

Comment:
looks slightly nicer to do `match *op { ...`, then you can do `Max` instead of `&Max`